service/src/main/java/bio/terra/landingzone/stairway/flight/delete/DeleteLandingZoneResourcesStep.java

```diff
@@ -69,6 +76,21 @@ public StepResult doStep(FlightContext context) throws RetryException {
           deletedLandingZone.landingZoneId(),
           deletedResources);
 
+    } catch (ManagementException e) {
+      // Azure returns AuthorizationFailed when an MRG is deleted or otherwise inaccessible. Since
+      // the user is unable to change the IAM permissions on an MRG due to deny assignments, we
```

Comment:
> the user is unable to change the IAM permissions on an MRG due to deny assignments
to make sure I'm following the full reasoning chain: since they have access to the landing zone, they must have access to the MRG, so if it fails it can't be due to a lack of permissions, and therefore must be because it's gone?

Reply:
The workspace manager client always has access to all Terra MRGs (since we are the publisher and have added that as an allowed principal for these sorts of operations). The _user_ (i.e., the billing project administrator or whoever), has access in that they can _see_ the MRG and see the resources inside but they cannot muck with permissions (aka IAM) and remove the WSM client's permissions. Therefore, if we get a 403, the MRG must be deleted or is otherwise permanently inaccessible.